index.html

```diff
@@ -1546,6 +1546,404 @@ <h3><dfn>close()</dfn> method</h3>
   </section>
 
+  <section data-dfn-for="TCPServerSocket">
+    <h2>
+      {{TCPServerSocket}} interface
+    </h2>
+
+    <pre class="idl">
+      [Exposed=(Window,Worker), SecureContext]
+      interface TCPServerSocket {
+        constructor(DOMString localAddress,
+                    optional TCPServerSocketOptions options = {});
+
+        readonly attribute Promise&lt;TCPServerSocketOpenInfo> opened;
+        readonly attribute Promise&lt;undefined> closed;
+
+        Promise&lt;undefined> close();
+      };
+    </pre>
+
+    <p>
+      Methods on this interface typically complete asynchronously, queuing work on the
+      <dfn>TCPServerSocket task source</dfn>.
+
+    <p>
+      Instances of {{TCPServerSocket}} are created with the internal slots described in the
+      following
+      table:
+
+    <table class="simple" data-dfn-for="TCPServerSocket">
+      <tr>
+        <th>Internal slot
+        <th>Initial value
+        <th>Description (non-normative)
+      <tr>
+        <td><dfn>[[\readable]]</dfn>
+        <td>`null`
+        <td>A {{ReadableStream}} used to accept incoming connections.
+      <tr>
+        <td><dfn>[[\openedPromise]]</dfn>
+        <td>`new Promise`
+        <td>A {{Promise}} used to wait for the socket to be opened. Corresponds to the
+          {{TCPServerSocket/opened}} member.
+      <tr>
+        <td><dfn>[[\closedPromise]]
+        <td>`new Promise`
+        <td>A {{Promise}} used to wait for the socket to close or error. Corresponds to the
+          {{TCPServerSocket/closed}} member.
+    </table>
+
+    <section>
+      <h3><dfn>constructor()</dfn> method</h3>
+
+      <aside class="example">
+        In order to accept incoming TCP connections a listening socket must be requested first.
+        The socket object constructor allows the site to specify necessary parameters which control
+        what interfaces the socket is supposed to listen on and how many connections can be queued
+        up.
+
+        Once {{TCPServerSocket/opened}} has resolved, {{TCPServerSocketOpenInfo/readable}}
+        can be accessed to get the {{ReadableStream}} instance to start accepting connections to the
+        server.
+      </aside>
+
+      The {{TCPServerSocket/constructor()}} steps are:
+
+      <ol>
+        <li>
+          If [=this=]'s [=relevant global object=]'s [=associated Document=] is
+          not [=allowed to use=] the [=policy-controlled feature=] named "[=policy-controlled
+          feature/direct-sockets=]", throw a "{{NotAllowedError}}" {{DOMException}}.
+        <li>
+          If |localAddress| is not a valid IP address, throw a {{TypeError}}.
+        <li>
+          If |options|["{{TCPServerSocketOptions/localPort}}"] is equal to 0, throw a {{TypeError}}.
+        <li>
+          If |options|["{{TCPServerSocketOptions/backlog}}"] is equal to 0, throw a {{TypeError}}.
+        <li>
+          If |options|["{{TCPServerSocketOptions/ipv6Only}}"] is set but `localAddress` is not equal
```

Comment:
```suggestion
          If |options|["{{TCPServerSocketOptions/ipv6Only}}"] is true but `localAddress` is not equal
```

It's okay for it to be set to false.